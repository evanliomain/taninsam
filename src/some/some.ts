/**
 * @module array=>boolean
 */
/**
 * Determines whether some of the members of an array satisfy the specified test.
 * @param iteratee A function that accepts up to three arguments.
 * @return the function to apply on the array to determine if its satisfy the specified test
 * @example
 * ```
 * some<number>(x => x < 3)([1, 2, 3, 4, 5]) // true
 * ```
 * @example Using the chain
 * ```
 * chain([1, 2, 3, 4, 5])
 *   .chain(some<number>(x => x < 3))
 *   .value() // true
 * ```
 */
export function some<T>(
  iteratee: (value: T, index: number, array: ReadonlyArray<T>) => boolean
): (array: ReadonlyArray<T>) => boolean {
  return (array: ReadonlyArray<T>) => array.some(iteratee);
}
