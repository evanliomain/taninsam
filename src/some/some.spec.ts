import { some } from './some';

describe('some function', () => {
  const input = [1, 2, 3, 4, 5];

  test('is a pure function', () => {
    some((x: number): boolean => x < 3)(input);
    expect(input).toMatchSnapshot();
  });

  test('[1, 2, 3, 4, 5] |> some === true', () => {
    expect(some((x: number): boolean => x < 3)(input)).toMatchSnapshot();
  });
});
