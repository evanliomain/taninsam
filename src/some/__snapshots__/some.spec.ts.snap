// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`some function [1, 2, 3, 4, 5] |> some === true 1`] = `true`;

exports[`some function is a pure function 1`] = `
Array [
  1,
  2,
  3,
  4,
  5,
]
`;
